README.md

```diff
@@ -72,12 +65,40 @@ A valid (read: that follows this interface) IPLD format implementation the follo
 
 `callback` must have the signature `function (err, dagNode)`, where `err` is an Error if the function fails and `dagNode` is the dagNode that got deserialized in the process.
 
-#### `util.cid(dagNode, callback)`
+#### `util.cid(dagNode[, options], callback)`
 
 > get the CID of the dagNode
 
+Options include:
+  - version - the CID version to be used (defaults to 1)
+  - hashAlg - the hash algorithm to be used (default to the one set by the format)
```

Comment:
Added these options so that we can pick the right hash function and not just default on the format (the format can always say that it only supports one hash function)

Reply:
I wouldn't say _assigned_ it is more. A dagNode is a thing in which we can calculate the CID of it, but it doesn't necessarily carry the CID as property.

The main reason why is because we want to have pure JSON blobs being first class dagNodes with CBOR and to add the CID in there I would have to wrap the JSON object into another object and that would be just a mess API wise.